### PR TITLE
Location feature

### LINKED ISSUE #1: Location Site
- fetch api (locataion)
- put data in component
- build page
- link page in router

### PR DESCRIPTION
**Pull request by annidrumm**
Add location-feature #1 
- solved issue
- copied PageNav from max episodes #7
- still some styling needed

### CODE REVIEW

**Review by mpagels — APPROVED** [review]

**Comment by maxbischof** [comment]
This pull request brings in a second version of cards styling. Lets discuss a general version. This shouldn't be a blocker for this pull request. Therefore I created an issue [here](https://github.com/neuefische/hh-2020-w4-rickandmorty-team1/issues/11).

File: src/components/pages/Locations/Location.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import styled from 'styled-components/macro'
+
+export default function Location({ name, type, dimension, residents }) {
```
Comment (maxbischof) [inline src/components/pages/Locations/Location.js]: Should we name the components Location and Episode? Or LocationCard and EpisodeCard? Right now we have mixed naming.

File: src/components/pages/Locations/LocationPage.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const [location, setlocation] = useState([])
+  const [pages, setPages] = useState()
+
+  useEffect(() => {
+    fetch('https://rickandmortyapi.com/api/location/')
+      .then((res) => res.json())
+      .then((data) => {
+        setlocation(data.results)
+        setPages(data.info.pages)
+      })
```
Comment (maxbischof) [inline src/components/pages/Locations/LocationPage.js]:
```suggestion
    fetchPageData(1)
```

File: src/components/pages/Locations/LocationPage.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }, [])
+  function fetchPageData(page) {
+    fetch('https://rickandmortyapi.com/api/location/?page=' + page)
+      .then((res) => res.json())
+      .then((data) => {
+        setlocation(data.results)
+      })
```
Comment (maxbischof) [inline src/components/pages/Locations/LocationPage.js]:
```suggestion
    fetch('https://rickandmortyapi.com/api/location/?page=' + page)
      .then((res) => res.json())
      .then((data) => {
        setlocation(data.results)
        setPages(data.info.pages)
      })
```

File: src/components/pages/Locations/LocationPage.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    <div>
+      {location.map((location, index) => (
+        <Location
+          key={index}
+          name={location.name}
+          type={location.type}
+          dimension={location.dimension}
+          residents={location.residents.length}
+        />
+      ))}
```
Comment (maxbischof) [inline src/components/pages/Locations/LocationPage.js]: I am always asking myself if its good practice to have logic in the return statement. What do you think? @annidrumm  @mpagels  @annaleaschroeder

**Review by maxbischof — CHANGES_REQUESTED** [review]
Thanks for the location feature. I added some feedback.

File: src/components/pages/Locations/Location.js (hunk 1 above)
Comment (annidrumm) [reply src/components/pages/Locations/Location.js]: Thanks for your feedback. And Good point. I prefer LocationCard as well and will update it today.

Comment (mpagels) [reply src/components/pages/Locations/Location.js]: LocationCard is way better for the understanding.

**Comment by annidrumm** [comment]
New issue:

- make it mobile friendly

**Pull request merged**